### PR TITLE
feat: disk cache switch to consistent hash and add bad disk detection

### LINKED ISSUE #4540: Enhance disk cache: consistent hash
**What would you like to be added**:
- add consistent hash support
- add disk cache uuid
- add bad disk detection
  - set free ratio to no less than 20% if in root volume

**Why is this needed**:
- consistent hashing supports the addition and removal of disk cache.
- disk cache's uuid allows for disk cache migration without affecting the results of consistent hashing.

### PR DESCRIPTION
**Pull request by jiefenghuang**
close #4540

### CODE REVIEW

File: pkg/chunk/disk_cache.go [hunk 1]
```diff
@@ -26,21 +26,29 @@ import (
...
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/twmb/murmur3"
 )
 
 var (
-	stagingDir = "rawstaging"
-	cacheDir   = "raw"
+	stagingDir   = "rawstaging"
+	cacheDir     = "raw"
+	maxIODur     = time.Second * 5
```
Comment (davies) [inline pkg/chunk/disk_cache.go:49]: 60 seconds

File: pkg/chunk/disk_cache.go [hunk 2]
```diff
@@ -797,46 +990,121 @@ func newCacheManager(config *Config, reg prometheus.Registerer, uploader func(ke
...
+		dir = s.dir
+	}
+	delete(m.storeMap, id)
+	for i, c := range m.stores {
+		if c != nil && c.id == id {
+			m.stores[i] = nil
+		}
+	}
+	m.Unlock()
+	logger.Infof("cache dir `%s`(%s) is unavailable, removed", dir, id)
```
Comment (davies) [inline pkg/chunk/disk_cache.go:1056]: error

File: pkg/chunk/disk_cache.go (hunk 1 above)
Comment (jiefenghuang) [reply pkg/chunk/disk_cache.go:49]: ok, read/write on hdd bad sector may hang for 30s.

may be we should retry for 3 ~ 5 times ?

Comment (jiefenghuang) [reply pkg/chunk/disk_cache.go:49]: change io timeout to 60s

File: pkg/chunk/disk_cache.go (hunk 2 above)
Comment (jiefenghuang) [reply pkg/chunk/disk_cache.go]: done

**Pull request merged**